pkg/cableengine/ipsec/strongswan.go

```diff
@@ -286,6 +296,43 @@ func (i *strongSwan) GetActiveConnections(clusterID string) ([]string, error) {
 	return connections, nil
 }
 
+// removeStaleCable removes cables that have no active SAS and returns a
+// boolean indicating removal of the cable. The status of the error is nil
+// in case of no errors, otherwise the returns values are false and the
+// error.
+func (i *strongSwan) removeStaleCable(client *goStrongswanVici.ClientConn, cableID string) (bool, error) {
+	sas, err := client.ListSas("", "")
+	if err != nil {
+		klog.Errorf("error while retrieving active sas")
+		return false, err
+	}
+
+	for _, samap := range sas {
+		klog.V(6).Infof("Found SA %v", samap)
+		sa, exists := samap[cableID]
+		if exists && (sa.State == "ESTABLISHED" || sa.State == "CONNECTING") {
+			// Cable with an active SA and thus not stale
+			return false, nil
+		}
+	}
+
+	klog.V(4).Infof("Remove stale cable: %s", cableID)
+	// Creating this stub Endpoint is not ideal but to avoid a refactor
+	// TODO(mpeterson): refactor code to make this not a requirement
+	err = i.DisconnectFromEndpoint(types.SubmarinerEndpoint{
+		Spec: subv1.EndpointSpec{
+			CableName: cableID,
+		},
+	})
+	if err != nil {
+		klog.Errorf("error while removing stale cable")
```

Comment:
... same here

Reply:
Whether we log here or propagate, can you also log `cableID` with the message for additional context? - not sure if that would be useful but can't hurt.